update document for tispark-2.2.0 release (#1094)
pingcap/tispark: docs/userguide_spark2.1.md

tisparkDF.write.saveAsTable("hive_table") // save table to hive
spark.sql("select * from hive_table a, tispark_table b where a.col1 = b.col1").show // join table across Hive and Tispark
```
## Load Spark DataFrame into TiDB using TiDB Connector

> TiSpark natively supports writing data to TiKV using Spark Data Source API. ACID is also guaranteed.

Reviewer: ```suggestion
TiSpark natively supports writing data to TiKV via Spark Data Source API and guarantees ACID.
```
or
```suggestion
TiSpark natively supports writing data to TiKV via Spark Data Source API and ACID guarantees.
```

Reviewer 2: 2.1 does not support native write right? Why here doc for 2.1 has this?

Author: fixed

For example:

```scala
val tidbOptions: Map[String, String] = Map(
  "tidb.addr" -> "127.0.0.1",
  "tidb.password" -> "",
  "tidb.port" -> "4000",
  "tidb.user" -> "root",
  "spark.tispark.pd.addresses" -> "127.0.0.1:2379"
)

val customer = spark.sql("select * from customer limit 100000")

customer.write
.format("tidb")
.option("database", "tpch_test")
.option("table", "cust_test_select")
.mode("append")
.save()
```

See [here](./datasource_api_userguide.md) for more details.

## Load Spark DataFrame into TiDB using JDBC

> Although TiSpark provides a direct way to load data into your TiDB cluster, you can still do it by using JDBC.

Reviewer: same here
